File: api/models/cinema.go
package models

import (
	"errors"
	"github.com/getsentry/sentry-go"
	"github.com/jinzhu/gorm"
)

type Cinema struct {
	gorm.Model

	Name      string   `json:"name"`
	City      string   `json:"city"`
	Company   string   `json:"company"`
	Latitude  *float64 `json:"latitude"`
	Longitude *float64 `json:"longitude"`
}

func SaveCinema(cinema *Cinema) error {
	db := db.Where(Cinema{Name: cinema.Name, City: cinema.City, Company: cinema.Company}).FirstOrCreate(&cinema)
	if db.Error != nil {
		sentry.CaptureException(db.Error)
		return errors.New("failed to create a cinema. " + db.Error.Error())
	}

	if cinema.ID <= 0 {
		sentry.CaptureException(errors.New("failed to create a cinema, connection error"))
		return errors.New("failed to create a cinema, connection error")
	}

	return nil
}

File: normalizer/data/Cinema.py
from dataclasses import dataclass


@dataclass
class Cinema:
    name: str
    city: str
    company: str
    longitude: float
    latitude: float

    def to_json(self):
        return {
            "name": self.name,
            "city": self.city,
            "company": self.company,
            "longitude": self.longitude,
            "latitude": self.latitude,
        }

    @staticmethod
    def from_json(j: dict):
        return Cinema(
            name=j["name"],
            city=j["city"],
            company=j["company"] if "company" in j else j["cinema"],
            longitude=j.get("longitude", None),
            latitude=j.get("latitude", None),
        )
